src/coreclr/jit/emitxarch.cpp

```diff
@@ -609,6 +609,54 @@ bool emitter::AreUpper32BitsZero(regNumber reg)
     return false;
 }
 
+//------------------------------------------------------------------------
+// AreUpper32BitsFFFF: check if some previously emitted
+//     instruction set the upper 32 bits of reg to 0xFFFF.
+//
+// Arguments:
+//    reg - register of interest
+//
+// Return Value:
+//    true if previous instruction upper 32 bits are 0xFFFF.
+//    false if it did not, or if we can't safely determine.
+bool emitter::AreUpper32BitsFFFF(regNumber reg)
```

Comment:
The name is potentially misleading, I think we want `AreUpper32BitsSignExtended` (or something similar) instead, since that's the "real ask".

e.g. `movsxd` on a value that was `0x7FFF_FFFF` will result in `0x0000_0000_7FFF_FFFF`, a secondary `movsxd` is then still redundant.

Reply:
Originally I did name it `SignExtended`, will switch it back.